src/sg/flybot/pullable.clj

```diff
@@ -64,16 +66,39 @@
       (ptn/->query (pattern->query f-named-var) pattern))
     {::compiled? true}))
 
+(defn compile-query
+  "If `query-or-pattern` is a pattern such as '{:a ?}, runs it to get the query,
+   If it is a query such as (query '{:a ?}), returns it."
+  [query-or-pattern]
+  (if (some-> query-or-pattern meta ::compiled?)
+    query-or-pattern
+    (query query-or-pattern)))
+
+(defn validate-data
+  "Runs the pulled-data against a `malli` schema.
+   Returns the pulled-data if it respects the schema, else throws error."
+  [pulled-data schema]
+  (if schema
+    (let [validator (m/validator schema)
+          data (first pulled-data)]
+      (if (validator data)
+        pulled-data
+        (throw
+         (let [err (m/explain schema data)]
+           (ex-info (str (me/humanize err))
+                    {:data pulled-data :error err})))))
+    pulled-data))
```

Comment:
This might be convenient for development but generally not a good practice for the production system:

 - emphasis on string message.
 - it could be slow.

Therefore, `explain-data` should be encouraged, especially if we will serialize it.